src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Atom10FeedFormatter.cs

```diff
@@ -871,6 +871,12 @@ private SyndicationFeed ReadFeedFrom(XmlReader reader, SyndicationFeed result, b
                                 }
                             }
                         }
+
+                        if (feedItems.Count != 0)
+                        {
+                            result.Items = feedItems;
+                        }
```

Comment:
Previously, if there were 0 items returned, result.Items would always be set, even to an empty collection. Now result.Items might end up being null.

Reply:
Fixed this.